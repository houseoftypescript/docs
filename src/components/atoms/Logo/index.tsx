import React from 'react';
import { CDN_IMAGE_URL } from '../../../constants';

export type LogoProps = { teamId: string; className?: string };

export const Logo: React.FC<LogoProps> = ({ teamId, className = '' }) => {
  const backgroundImage: string = [
    `url(${CDN_IMAGE_URL}/teams/png/${teamId}.png)`,
    `url(${CDN_IMAGE_URL}/teams/jpg/${teamId}.jpg)`,
  ].join(', ');

  return (
    <div
      className={`bg-cover bg-center ${className}`}
      style={{ backgroundImage }}
    />
  );
};

Logo.displayName = 'Logo';
Logo.defaultProps = { className: '' };

export default Logo;
