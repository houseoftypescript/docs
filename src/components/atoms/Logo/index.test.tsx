import Logo from '.';
import { render } from '@testing-library/react';

describe('Logo', () => {
  test('render default', () => {
    const wrapper = render(<Logo teamId="teamId" />);
    expect(wrapper.container).toMatchSnapshot();
  });
});
