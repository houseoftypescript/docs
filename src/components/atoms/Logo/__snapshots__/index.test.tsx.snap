// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo render default 1`] = `
<div>
  <div
    class="bg-cover bg-center "
  />
</div>
`;
